### PR TITLE
Add migration from FMP to RDO

### PR DESCRIPTION
**Pull request by Wolvan**
After some thinking, we decided to make FixMyPosts automatically install
Reblog Display Options, due to FMP being rather buggy and tons of people
asking for support for it. Permission has been granted by @cunfricht

### CODE REVIEW

File: Extensions/FixMyPosts.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+//* DESCRIPTION Fixes the reblog format of posts to reflect the blockquote-style **//
+//* DEVELOPER THETIMEISNEVERIGHT **//
+//* FRAME false **//
+//* BETA false **//
+
+XKit.extensions.FixMyPosts = new Object({
+
+    running: false,
+
+    preferences: {
```
Comment (nightpool) [inline Extensions/FixMyPosts.js]: should have the existing FMP preferences so we can migrate some of this stuff.

File: Extensions/FixMyPosts.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
+//* TITLE FixMyPosts **//
+//* VERSION 999.999.999 **//
```
Comment (nightpool) [inline Extensions/FixMyPosts.js]: just 2.0.0 is fine.

File: Extensions/FixMyPosts.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+//* DESCRIPTION Fixes the reblog format of posts to reflect the blockquote-style **//
+//* DEVELOPER THETIMEISNEVERIGHT **//
```
Comment (nightpool) [inline Extensions/FixMyPosts.js]: lowercase

File: Extensions/FixMyPosts.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            text: "Please use Reblog Display Options instead of FixMyPosts",
+            type: "separator"
+        },
+    },
+
+    run: function() {
+        this.running = true;
+
+        XKit.window.show("Use Reblog Display Options instead of FixMyPosts",
+            "<p>FixMyPosts is rather buggy and a third party extension, "+
```
Comment (nightpool) [inline Extensions/FixMyPosts.js]: messaging needs some work tbh.

File: Extensions/FixMyPosts.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+            "not supported by the New XKit Team. Reblog Display Options has"+
+			"the same functionality and is an official extension. You"+
+			"just need to set RDO's style to nested in the RDO preferences.</p>"+
```
Comment (nightpool) [inline Extensions/FixMyPosts.js]: should set nested transparently.

File: Extensions/FixMyPosts.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+            '<div id="xkit-close-message" class="xkit-button">Don\'t install</div>');
+        $("#xkit-install-br").click(function(){
+            XKit.install("better_reblogs", function(mdata) {
+                if (mdata.errors) {
+                    if (mdata.storage_error === true) {
+                        show_error_installation("[Code: 631] Can't store data on browser");
+                        return;
+                    }
+                    if (mdata.server_down === true) {
+                        show_error_installation("[Code: 101] Can't reach XKit servers");
```
Comment (nightpool) [inline Extensions/FixMyPosts.js]: should both specifically reference "error installing RDO" (and link to new-xkit-support? not sure if show_error_installation does this or not)

File: Extensions/FixMyPosts.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+                        if (mdata.file === "not_found") {
+                            show_error_installation("Can't download " + to_install + ": Not found");
+                        } else {
+                            show_error_installation("Can't download " + to_install);
+                        }
+                    }
+                    return;
+                }
+
+                XKit.installed.remove("FixMyPosts");
```
Comment (nightpool) [inline Extensions/FixMyPosts.js]: should set config value for nested here

File: Extensions/FixMyPosts.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+        XKit.window.show("Use Reblog Display Options instead of FixMyPosts",
+            "<p>FixMyPosts is rather buggy and a third party extension, "+
+            "not supported by the New XKit Team. Reblog Display Options has"+
+			"the same functionality and is an official extension. You"+
+			"just need to set RDO's style to nested in the RDO preferences.</p>"+
+            "<br>Would you like to install Reblog Display Options?",
+            "info",
+            '<div id="xkit-install-br" class="xkit-button default">Install Reblog Display Options</div>'+
+            '<div id="xkit-close-message" class="xkit-button">Don\'t install</div>');
```
Comment (nightpool) [inline Extensions/FixMyPosts.js]: should be clear that FMP won't work after this.

**Comment by nightpool** [comment]
general notes: should also add some config options to RDO that match some
of the FMP options, to make users happy if nothing else

**Comment by Wolvan** [comment]
@nightpool I haven't checked what options FMP has, I thought it just... you know... exists? I'll look into it. How would we best go about setting RDO's options?

Also, could you make a suggestion for the text? I ain't the best at english as you know

File: Extensions/FixMyPosts.js (hunk 3 above)
Comment (dlmarquis) [inline Extensions/FixMyPosts.js]: It's all-caps in the original extension. I don't think there's a good reason to change it at this point?

**Comment by hobinjk** [comment]
FMP's preferences are ```javascript
		"thick_quote" : {
			text: "Make blockquotes thicker and darker",
			default: false,
			value: false
		},
		"condense_quote" : {
			text: "Condense block quotes",
			default: true,
			value: true
		},
		"alert_on_update" : {
			text: "Display alert in top right corner when update is available (might be buggy!)",
			default: true,
			value: true
		}
```

**Comment by Wolvan** [comment]
@hobinjk I know by now, but I don't think RDO has any tweaks that are equal to these settings, right?

**Pull request closed without merging**